internal/venice-common/src/main/java/com/linkedin/venice/pubsub/PubSubPositionType.java

```diff
@@ -0,0 +1,8 @@
+package com.linkedin.venice.pubsub;
+
+// Do not change the values of these constants.
+public class PubSubPositionType {
+  public static final int APACHE_KAFKA_OFFSET = 0;
+  public static final int LI_INTERNAL_XC = 1;
+  public static final int LI_INTERNAL_RESERVED = 2;
+}
```

Comment:
Hmm... this doesn't seem right. If there is a need for such mapping, and the mapping needs to support internal extensions, then we shouldn't let it be hardcoded. I can see two options which seem clean:

1. The mapping is fully dynamic, e.g. coming from config or perhaps from the controller (exposed via endpoint, via some state in ZK, or whatever).
2. The mapping is hard-coded for pub subs which are available in open source, e.g. starting from zero and going up from there, and furthermore there is an ability to inject internal mappings via whichever mechanism is chosen to achieve (1), e.g. occupying negative numbers.

Reply:
> If there is a need for such mapping
I thinking if we can use magic byte to achieve the same effect

Reply:
As per our discussion, using the `type` field in `PubSubPositionWireFormat` to include the above info.